### PR TITLE
Deprecate IContainerRuntimeOptions.flushMode

### PR DESCRIPTION
**Pull request by markfields**
## Description

Fixes [AB#26497](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/26497)

Only the default value `FlushMode.TurnBased` is supported when calling `ContainerRuntime.loadRuntime` directly, so there's no need for consumers to pass this option in.  We'll remove the option altogether in `2.20.0`.

## Reviewer Guidance

Note that `FlushMode.Immediate` is still used in `compatibilityConfiguration.ts`, for 1.x compat mode.  #23272 adds an internal-only set of options which includes `flushMode`, to support that use.  But in the meantime we can finish removing it from the legacy-alpha API surface.

### CODE REVIEW

**Review by jatgarg — APPROVED** [review]

**Comment by markfields** [comment]
@jzaffiro -- want to review this changeset?  Similar to #23260

**Review by kian-thompson — APPROVED** [review]

File: .changeset/fancy-insects-sing.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+"@fluidframework/container-runtime": minor
+---
+---
+"section": deprecation
+---
+
+Deprecating `IContainerRuntimeOptions.flushMode`
```
Comment (tylerbutler) [inline .changeset/fancy-insects-sing.md:8]:
```suggestion
IContainerRuntimeOptions.flushMode is now deprecated
```

File: .changeset/fancy-insects-sing.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+Only the default value `FlushMode.TurnBased` is supported when calling `ContainerRuntime.loadRuntime` directly,
+so there's no need for consumers to pass this option in.  We'll remove the option altogether in `2.20.0`.
```
Comment (tylerbutler) [inline .changeset/fancy-insects-sing.md:11]:
```suggestion
The `IContainerRuntimeOptions.flushMode` property is deprecated and will be removed in version 2.20.0. 

Only the default value `FlushMode.TurnBased` is supported when calling `ContainerRuntime.loadRuntime` directly, so there's no need for consumers to pass this option in.
```

**Comment by markfields** [comment]
@tylerbutler thanks for the feedback, I took your suggestions.

**Review by alexvy86 — APPROVED** [review]

**Pull request merged**